### PR TITLE
Component/dropdown menu

### PR DESCRIPTION
**Pull request by AngelikaWieck**
# Issue #15 - Dropdown Menüs 

## Description

implemented the DropdownMenu-Component

Im Storybook zeige ich die Komponente unter den "Base Components" an, weil ich nicht wusste, was ein besserer Ort dafür ist (die DropdownMenu-Komponente ist ja eigtl. keine "Base Component"). Also müsste das noch gefixed werden, wenn es einen besseren Ort gibt :) 

## Screenshot
image

### CODE REVIEW

File: src/components/DropdownMenu.vue [hunk 1]
```diff
@@ -0,0 +1,72 @@
+<template>
+	<div class="dropdown">
+		<button class="button">{{ title }}</button>
```
Comment (adrianjost) [inline src/components/DropdownMenu.vue]: Verwende bitte `BaseButton` statt Button und wrappe diesen in einen `slot`, damit man das Dropdown auch mit anderen Elementen verwenden kann.

**Comment by adrianjost** [comment]
I found some inspiration: “Paying Attention to Details” by Shlok Desai https://medium.com/p/eb0c43d0b02b

File: src/components/uidMixin.js [hunk 2]
```diff
@@ -0,0 +1,9 @@
+let uid = 0;
```
Comment (adrianjost) [inline src/components/uidMixin.js]: @smyth64 I like the idea of using mixins, but I don't think they belong into the `components` folder. WIth your Nuxt experience, what place do you recommend?

Normally I would create a seperated `src/mixins`-folder for such stuff to keep the components folder clean and consistency filles with vue and unit test files only.

Comment (adrianjost) [reply src/components/uidMixin.js]: @AngelikaWieck After merging this pull request, could you implement your elegant solution [here](https://github.com/schul-cloud/nuxt-client/blob/master/src/components/ui/BaseInput.vue#L14) as well so we can remove the prop `id` there and be more consistent?

Comment (AngelikaWieck) [reply src/components/uidMixin.js]: @adrianjost It is even easier:
```
<input id="foo" />
<label for="foo">Bar</label>
```

...can be written as:
```
<label>
  <input />
  Bar
</label>
```

Maybe `Bar` could be wrapped in a span for styling.

So, no `id` is needed  🎉

Comment (adrianjost) [reply src/components/uidMixin.js]: @AngelikaWieck You may be right, but @dominikbruechner encountered some problems using <label> to wrap the `FlatPickr`-Component ([Link](https://github.com/schul-cloud/nuxt-client/blob/master/src/components/ui/BaseInput.vue#L43)). We should check this again. I will create some Issues for that

**Comment by dominikbruechner** [comment]
Could you please add screenshots in specific UI relevant PRs? cheers :)

File: stories/misc.stories.js [hunk 3]
```diff
@@ -25,9 +25,9 @@ storiesOf("Misc", module)
 		components: { DropdownMenu, MenuLink },
 		template: `
 <DropdownMenu title="Dropdown">
-	<MenuLink href="#" class="test">Link 101</MenuLink>
-	<MenuLink href="#">Link 2</MenuLink>
-	<MenuLink href="#">Link 3</MenuLink>
+	<MenuLink href="/">Link 1</MenuLink>
```
Comment (adrianjost) [inline stories/misc.stories.js]: Check your browser console warnings. I think you need to use `to` instead of `href` for internal links.

**Pull request merged**